### PR TITLE
BOAC-1705, vue and vuex upgrade; stricter linting per vuejs.org style-guide

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1705

Vue template linting as prescribed in https://vuejs.org/v2/style-guide/  I removed 'prettier' due to eslint rule conflicts.  Review with `?w=1`

### CODE REVIEW

File: src/components/HeaderMenu.vue [hunk 1]
```diff
@@ -3,11 +3,12 @@
...
       <span v-if="user && !user.firstName">{{ user }}</span>
-      <b-dropdown id="header-dropdown-under-name"
-                  class="mr-3"
-                  variant="link"
-                  no-caret
-                  right>
+      <b-dropdown
+        id="header-dropdown-under-name"
+        class="mr-3"
+        variant="link"
```
Comment (johncrossman) [inline src/components/HeaderMenu.vue]: The indentation rule above is the most significant change to our code style.

Comment (paulkerschen) [reply src/components/HeaderMenu.vue]: That's definitely an easier indentation rule to deal with, especially when the element names get long.

File: .eslintrc.js [hunk 2]
```diff
@@ -5,23 +5,27 @@ module.exports = {
...
   },
   extends: [
-    '@vue/prettier',
     '@vue/typescript',
-    'eslint:recommended',
-    'plugin:vue/essential'
+    'plugin:vue/recommended',
+    'eslint:recommended'
   ],
-  plugins: ['vue', 'prettier'],
```
Comment (paulkerschen) [inline .eslintrc.js]: This is subjective, but I didn't find `prettier` to be all that pretty and don't think I'll miss it.

**Comment by paulkerschen** [comment]
Thanks for this, @johncrossman - these seem like sensible rules. Given the size of the diff, I'll give someone else the chance to second.

**Comment by johncrossman** [comment]
Thank you @paulkerschen.  (Maybe @lyttam is available as a second pair of eyes.)

**Comment by lyttam** [comment]
Looks good, thanks @johncrossman!

**Pull request merged**